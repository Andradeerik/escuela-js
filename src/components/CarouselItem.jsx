import React from 'react';
import PropTypes from 'prop-types';
import '../assets/styles/components/CarouselItem.scss';
import playIcon from '../assets/static/play.png';

const CarouselItem = ({ cover, title, year, contentRating, duration }) => (
  <div className="carousel-item">
    <img className="carousel-item__img" src={cover} alt={title} />
    <div className="carousel-item__details">
      <img
        className="carousel-item__details--img"
        src={playIcon}
        alt="play-icon"
      />
      <p className="carousel-item__details--title">{title}</p>
      <p className="carousel-item__details--subtitle">
        {year}
        {' '}
        {contentRating}
        {' '}
        {duration}
      </p>
    </div>
  </div>
);

CarouselItem.propTypes = {
  title: PropTypes.string,
  year: PropTypes.number,
  contentRating: PropTypes.string,
  duration: PropTypes.number,
  cover: PropTypes.string,
};

export default CarouselItem;
