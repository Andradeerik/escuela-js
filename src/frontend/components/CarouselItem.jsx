import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setFavorite } from '../actions';
import '../assets/styles/components/CarouselItem.scss';
import playIcon from '../assets/static/play.png';
import addIcon from '../assets/static/add.png';

const CarouselItem = (props) => {
  const { cover, title, year, contentRating, duration, isList } = props;
  const handleSetFavorite = () => {
    props.setFavorite({
      cover, title, year, contentRating, duration,
    });
  };
  return (
    <div className='carousel-item'>
      <img className='carousel-item__img' src={cover} alt={title} />
      <div className='carousel-item__details'>
        <img
          className='carousel-item__details--img'
          src={playIcon}
          alt='play-icon'
        />
        {!isList ? (
          <div
            className='carousel-item__details--subtitle'
            onClick={handleSetFavorite}
            role='button'
            tabIndex='0'
          >
            <img
              className='carousel-item__details--img'
              src={addIcon}
              alt='play-icon'
            />
          </div>
        ) : null
        }
        <p className='carousel-item__details--title'>{title}</p>
        <p className='carousel-item__details--subtitle'>
          {year}
          {' '}
          {contentRating}
          {' '}
          {duration}
        </p>
      </div>
    </div>
  );
};

CarouselItem.propTypes = {
  title: PropTypes.string,
  year: PropTypes.number,
  contentRating: PropTypes.string,
  duration: PropTypes.number,
  cover: PropTypes.string,
};

const mapDispatchToProps = {
  setFavorite,
};

export default connect(null, mapDispatchToProps)(CarouselItem);
