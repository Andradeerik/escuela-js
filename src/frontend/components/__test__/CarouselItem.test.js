import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import CarouselItem from '../CarouselItem';

describe('CarouselItem component', () => {
  test('Matches the snapshot', () => {
    const carousel = create(
      <ProviderMock>
        <CarouselItem />
      </ProviderMock>,
    );
    expect(carousel.toJSON()).toMatchSnapshot();
  });
});
