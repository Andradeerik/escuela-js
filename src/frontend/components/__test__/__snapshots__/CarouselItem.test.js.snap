// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem component Matches the snapshot 1`] = `
<div
  className="carousel-item"
>
  <img
    className="carousel-item__img"
  />
  <div
    className="carousel-item__details"
  >
    <div>
      <a
        href="/player/undefined"
        onClick={[Function]}
      >
        <img
          alt="Play Icon"
          className="carousel-item__details--img"
          src="test-file-stub"
        />
      </a>
      <img
        alt="Reproducir"
        className="carousel-item__details--img"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <p
      className="carousel-item__details--title"
    />
    <p
      className="carousel-item__details--subtitle"
    >
      undefined undefined undefined
    </p>
  </div>
</div>
`;
